tools/common.py

```diff
@@ -1968,10 +1978,31 @@ def add_principled_shader(mesh):
 
                 # This is an image node, so link it to the principled shader later
                 node_image = node
-
-            if not node_image or node_image_count > 1:
+            #this material doesn't have a texture and doesn't have a MMD AO+Diffuse so skip
+            if (not node_image or node_image_count > 1) and not needsmmdcolor:
                 continue
-
+            elif needsmmdcolor and node_mmd_shader: #this needs to implement mmd color and has a shader node
+                #bake AO and Diffuse color into pixels for MMD texture. if texture exists, multiply over
+                #Thank this guy for pixel manipulation: https://blender.stackexchange.com/a/652
+                
+                basecolor = [max(0,min(1, a + b )) for a, b in zip(node_mmd_shader.inputs[0].default_value[:], [x*0.6 for x in node_mmd_shader.inputs[1].default_value[:]])]
```

Comment:
This seems unnecessarily complex, there is no need to force all this into one line

Reply:
I could do it in for loops, lemme try that